src/connectedk8s/azext_connectedk8s/custom.py

```diff
@@ -2046,3 +2047,193 @@ def get_custom_locations_oid(cmd, cl_oid):
+
+        # Setting kube_config
+        kube_config = set_kube_config(kube_config)
+        kube_client.rest.logger.setLevel(logging.WARNING)
+
+        # Loading the kubeconfig file in kubernetes client configuration
+        load_kube_config(kube_config, kube_context)
+        configuration = kube_client.Configuration()
+
+        # Install helm client
+        helm_client_location = install_helm_client()
+
+        # Install kubectl client
+        kubectl_client_location = install_kubectl_client()
+        release_namespace = validate_release_namespace(client, cluster_name, resource_group_name, configuration, kube_config, kube_context, helm_client_location)
+
+        # Checking the connection to kubernetes cluster.
+        # This check was added to avoid large timeouts when connecting to AAD Enabled AKS clusters
+        # if the user had not logged in.
+        check_kube_connection(configuration)
+        utils.try_list_node_fix()
+
+        # Fetch Connected Cluster for agent version
+        connected_cluster = get_connectedk8s(cmd, client, resource_group_name, cluster_name)
+
+        # Creating timestamp folder to store all the diagnoser logs
+        current_time = time.ctime(time.time())
+        time_stamp = ""
+        for elements in current_time:
+            if(elements == ' '):
+                time_stamp += '-'
+                continue
+            elif(elements == ':'):
+                time_stamp += '.'
+                continue
+            time_stamp += elements
+        time_stamp = cluster_name + '-' + time_stamp
+        # Generate the diagnostic folder in a given location
+        filepath_with_timestamp, diagnostic_folder_status = troubleshootutils.create_folder_diagnosticlogs(time_stamp)
+        if(diagnostic_folder_status != consts.Folder_Created):
+            storage_space_available = False
+
+        # To store the connected cluster resource logs in the diagnostic foler
+        diagnostic_checks["fetch_connected_cluster_resource"], storage_space_available = troubleshootutils.fetch_connected_cluster_resource(filepath_with_timestamp, connected_cluster, storage_space_available)
+        corev1_api_instance = kube_client.CoreV1Api(kube_client.ApiClient(configuration))
+
+        # Check if agents have been added to the cluster
+        arc_agents_pod_list = corev1_api_instance.list_namespaced_pod(namespace="azure-arc")
+
+        # To verify if arc agents have been added to the cluster
+        if arc_agents_pod_list.items:
+
+            # For storing all the agent logs using the CoreV1Api
+            diagnostic_checks["retrieved_arc_agents_logs"], storage_space_available = troubleshootutils.retrieve_arc_agents_logs(corev1_api_instance, filepath_with_timestamp, storage_space_available)
```

Comment:
all these 7-8 checks names should go into constants.py and we should refer using that :) . Please modify that for all the below lines as well